### PR TITLE
named_type_impl: Add star operator for access to inner value.

### PR DESCRIPTION
**Pull request by Shachlan**

### CODE REVIEW

**Comment by Shachlan** [comment]
@StatusReport

File: named_type_impl.hpp [hunk 1]
```diff
@@ -53,6 +53,19 @@ class FLUENT_EBCO NamedType : public Skills<NamedType<T, Parameter, Skills...>>.
         argument& operator=(argument &&) = delete;
     };
     
+    constexpr const T &operator*() const & {
```
Comment (StatusReport) [inline named_type_impl.hpp:56]: I'm probably missing some modern C++ syntax. What's `const &`? Same below.

Comment (Shachlan) [reply named_type_impl.hpp:56]: const after function declaration - means that the function doesn't modify the state of the object.

& after function declaration - means that the function is declared only for lvalue references.
&& after function declaration - same, for rvalue references.

File: named_type_impl.hpp [hunk 2]
```diff
@@ -53,6 +53,19 @@ class FLUENT_EBCO NamedType : public Skills<NamedType<T, Parameter, Skills...>>.
...
+      return value_;
+    }
+    constexpr T &operator*() & {
+      return value_;
+    }
+    constexpr const T &&operator*() && {
```
Comment (StatusReport) [inline named_type_impl.hpp]: Add `const`

Comment (Shachlan) [reply named_type_impl.hpp]: done

**Comment by Shachlan** [comment]
Finished squashing.

**Pull request merged**